test/Input/PathParamTest.php

```diff
@@ -96,7 +95,7 @@ public function testValidatorRaisesExceptionIfValueIsNonExistentPathAndMustExist
         $this->param->setPathMustExist(true);
         $validator = $this->param->getQuestion()->getValidator();
 
-        $this->expectException(RuntimeException::class);
+        $this->expectException(InvalidArgumentException::class);
```

Comment:
Since changing the exception type doesn't seem to be a big BC break/issue, perhaps all other `throw new RuntimeException...` can also be replaced by assertions?

Reply:
There's a few places where we get more specific and assertions don't work (I looked into it). I think at this point, I've used assertions wherever I can, but I'll review again once I've done other changes.